UKHO.MaritimeSafetyInformation.Web/UKHO.MaritimeSafetyInformation.Web/Services/NMDataService.cs

```diff
@@ -49,55 +49,51 @@ public async Task<List<ShowFilesResponseModel>> GetWeeklyBatchFiles(int year, in
+        public async Task<List<YearWeekModel>> GetAllYearWeek(string correlationId)
         {
-            List<KeyValuePair<string, string>> weeks = new();
+            List<YearWeekModel> yearWeekModelList = new();
+            try
+                {
+                string accessToken = await _authFssTokenProvider.GenerateADAccessToken(correlationId);
 
-            _logger.LogInformation(EventIds.GetAllWeeksOfYearStarted.ToEventId(), "Maritime safety information request to get all weeks of year to populate week dropdown started", correlationId);
+                _logger.LogInformation(EventIds.GetSearchAttributeRequestDataStarted.ToEventId(), "Request Search Attribute Year and week data from File Share Service started for _X-Correlation-ID:{correlationId}", correlationId);
 
-            weeks.Add(new KeyValuePair<string, string>("Week Number", ""));
+                IResult<BatchAttributesSearchResponse> searchAttributes = await _fileShareService.FssSearchAttributeAsync(accessToken, correlationId);
 
-            DateTimeFormatInfo dateTimeFormatInfo = DateTimeFormatInfo.CurrentInfo;
-            DateTime lastdate;
-            if (DateTime.Now.Year == year)
-            {
-                lastdate = new DateTime(year, DateTime.Now.Month, DateTime.Now.Day);
-            }
-            else
-            {
-                lastdate = new DateTime(year, 12, 31);
+                if (searchAttributes.Data != null)
+                {
+                    for (int i = 0; i < searchAttributes.Data.BatchAttributes.Count; i++)
+                    {
+                        if (searchAttributes.Data.BatchAttributes[i].Key == "YEAR / WEEK")
+                        {
+                            List<string> yearWeekList = searchAttributes.Data.BatchAttributes[i].Values;
+
+                            if (yearWeekList != null && yearWeekList.Count != 0)
+                            {
+                                foreach (string yw in yearWeekList)
+                                {
+                                    string[] yearWeek = yw.Split('/');
+                                    yearWeekModelList.Add(new YearWeekModel { Year = yearWeek[0].Trim(), Week = yearWeek[1].Trim() });
+                                }
+                                _logger.LogInformation(EventIds.GetSearchAttributeRequestDataFound.ToEventId(), "Request Search Attribute Year and week data recieved successfully from File Share Service for BatchSearchAttribute with _X-Correlation-ID:{correlationId}" , correlationId);
+                            }
+                            else
+                            {
+                                _logger.LogInformation(EventIds.GetSearchAttributeRequestDataNotFound.ToEventId(), "No Data recieved from File Share Service for Request Search Attribute Year and week for _X-Correlation-ID:{correlationId}", correlationId);
+                            }
```

Comment:
Please break the outer for loop if `YEAR / WEEK` is found.

Reply:
done

Reply:
Please can you use  class level constant for it.